<a href="https://colab.research.google.com/github/ShivaKottha/Simulation-and-Modeling/blob/main/homework5.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

The states



In [1]:
import numpy as np

Activity_state = dict(zip(range(3), ['rest', 'sweeties', 'exercises']))
print(Activity_state)


{0: 'rest', 1: 'sweeties', 2: 'exercises'}


The following describes the given state diagram's transition matrix:

In [2]:
import numpy as np

transition_matrix = np.array([[0.2, 0.2, 0.6],
                             [0.2, 0.1, 0.7],
                             [0.1, 0.3, 0.6]])

print(transition_matrix)


[[0.2 0.2 0.6]
 [0.2 0.1 0.7]
 [0.1 0.3 0.6]]


1. The starting state is Rest. Calculate the probability of possible activity on the 15th day.
2. What about the state and probability after 10000 days?

In [3]:
import numpy as np

# Transition matrix
transition_matrix = np.array([[0.2, 0.2, 0.6],
                             [0.2, 0.1, 0.7],
                             [0.1, 0.3, 0.6]])

# Starting state vector
start_state = np.array([1, 0, 0])  # Rest is the starting state

# Define the number of days
num_days = 15

# Calculate the probability after num_days days
prob_num_days = np.linalg.matrix_power(transition_matrix.T, num_days) @ start_state

# Print the probability after num_days days
print(f"Probability of activities on the {num_days}th day:")
activities = ['Rest', 'Sweeties', 'Exercises']
for activity, prob in zip(activities, prob_num_days):
    print(f"{activity}: {prob}")

# Define the number of days
num_days = 10000

# Calculate the probability after num_days days
prob_num_days = np.linalg.matrix_power(transition_matrix.T, num_days) @ start_state

# Print the probability after num_days days
print(f"\nProbability of activities after {num_days} days:")
for activity, prob in zip(activities, prob_num_days):
    print(f"{activity}: {prob}")


Probability of activities on the 15th day:
Rest: 0.13761467889913792
Sweeties: 0.23853211009159786
Exercises: 0.6238532110092637

Probability of activities after 10000 days:
Rest: 0.1376146788990357
Sweeties: 0.23853211009166178
Exercises: 0.6238532110089616


3. What do you observe from the above two?


Over an extended period, such as 10,000 days, a noticeable trend emerges: the probabilities of different outcomes tend to remain consistent. This observation suggests that the distribution has reached a stable state. Strikingly, the probabilities after 10,000 days closely resemble those after only 15 days. As the number of days increases, the probability values for each state gradually converge. Notably, when the number of trials is small, the model tends to assign disproportionately high odds to the "rest" and "sweets" categories. However, as the number of trials rises, the probabilities settle at approximately 13.76% for "rest," 23.85% for "sweets," and 62.39% for "exercises."